src/test/java/de/tum/in/ase/parser/IntegrationTest.java

```diff
@@ -32,6 +32,8 @@
 
+    private static final String EXPECTED_GCC = "{\"tool\":\"GCC\",\"issues\":[{\"filePath\":\"ascii_table.c\",\"startLine\":7,\"endLine\":7,\"startColumn\":13,\"endColumn\":13,\"rule\":\"[-Wunused-but-set-variable]\",\"category\":\"Misc\",\"message\":\"[-Wunused-but-set-variable]: variable ‘arr’ set but not used \\n        7 |         int arr[5];\\n        |             ^~~\\n        ascii_table.c: In function ‘decimal_to_hexadecimal’:\",\"priority\":\"warning\"},{\"filePath\":\"ascii_table.c\",\"startLine\":32,\"endLine\":32,\"startColumn\":15,\"endColumn\":15,\"rule\":\"[-Wimplicit-function-declaration]\",\"category\":\"Misc\",\"message\":\"[-Wimplicit-function-declaration]: implicit declaration of function ‘malloc’ \\n        32 |         ptr = malloc(sizeof(int));\\n        |               ^~~~~~\",\"priority\":\"warning\"},{\"filePath\":\"ascii_table.c\",\"startLine\":32,\"endLine\":32,\"startColumn\":15,\"endColumn\":15,\"rule\":\"[-Wbuiltin-declaration-mismatch]\",\"category\":\"Misc\",\"message\":\"[-Wbuiltin-declaration-mismatch]: incompatible implicit declaration of built-in function ‘malloc’ \\n        32 |         ptr = malloc(sizeof(int));\\n        |               ^~~~~~\",\"priority\":\"warning\"},{\"filePath\":\"ascii_table.c\",\"startLine\":31,\"endLine\":31,\"startColumn\":14,\"endColumn\":14,\"rule\":\"[-Wunused-but-set-variable]\",\"category\":\"Misc\",\"message\":\"[-Wunused-but-set-variable]: variable ‘ptr’ set but not used \\n        31 |         int *ptr;\\n        |              ^~~\",\"priority\":\"warning\"},{\"filePath\":\"ascii_table.c\",\"startLine\":40,\"endLine\":40,\"startColumn\":16,\"endColumn\":16,\"rule\":\"[-Wanalyzer-malloc-leak]\",\"category\":\"Memory\",\"message\":\"[-Wanalyzer-malloc-leak]: leak of ‘ptr’ [CWE-401] \\n        40 |         return hex_num;\\n        |                ^~~~~~~\\n        ‘decimal_to_hexadecimal’: events 1-4\\n        |\\n        |   32 |         ptr = malloc(sizeof(int));\\n        |      |               ^~~~~~~~~~~~~~~~~~~\\n        |      |               |\\n        |      |               (1) allocated here\\n        |   33 |\\n        |   34 |         for (int i = 0; dec_num != 0; i++) {\\n        |      |                         ~~~~~~~~~~~~\\n        |      |                                 |\\n        |      |                                 (2) following ‘false’ branch (when ‘dec_num == 0’)...\\n        |......\\n        |   40 |         return hex_num;\\n        |      |                ~~~~~~~\\n        |      |                |\\n        |      |                (3) ...to here\\n        |      |                (4) ‘ptr’ leaks here; was allocated at (1)\\n        |\\n\",\"priority\":\"warning\"}]}";
```

Comment:
Wouldn't it possible to store this string in a file like it's done for the `gcc.xml`?

Reply:
I refactored the tests to mainly use files instead of strings. But I am not sure how I feel about the change. It adds complexity to the code. But of course it makes the strings a bit nicer to work with as we don't have to escape them. What do you think of my changes?

Reply:
Far cleaner now. In my eyes this looks better. Especially now I don‘t have to scroll a long way any more to skip the strings.